blog/content/post/2020-01-15-GRUB2_and_3mdeb_minisummit.md

```diff
@@ -0,0 +1,177 @@
+  citizen in GRUB2 bootloader
+* [AMD TrenchBoot support in GRUB2](https://cloud.3mdeb.com/index.php/s/SjLdXJomaS6obYH) - final talk in
+  which I present status of AMD TrenchBoot support, implementation and possible paths to
+  upstream related code
+
+We also published all videos on our [YouTube channel]().
+
+# Redundant GRUB2 env file
+
+Key problem mentioned during presentations are lack of integrity check in
+current implementation of GRUB2, what means any corruption of environment file
+in GRUB may lead to boot failure that can be recovered only by manual
+intervention. Since in most cases GRUB2 is used as bootloader in general
+purpose computers this was not a problem. Even if in reality it happen people
+know how to recover either through GRUB command line or live booting Linux
+distro.
+
+Unfortunately, if trying to use GRUB2 in embedded environment there is risk of
+ending up with unbootable product, which of course leads to all other problems
+on vendor side.
+
+We all agreed that this feature is needed in GRUB2 and if 3mdeb will get
+correct sponsoring from vendors using GRUB2 such support will be developed and
+contributed. On the other hand some vendors assume that file corruption is so
+rare that they don't want to spend additional engineering hours on such
+support.
+
+Before such support is available there are still methods to mitigate impact of
+the problem for which reference you can find in presentation.
+
+# TPM support in GRUB2 for legacy boot mode
+
+Key problem here is lack of boot process integrity in non-UEFI systems. Since
+GRUB2 is in boot process chain also in legacy systems we wanted to continue
+discussion started at LPC 2019. General conclusion on Linux Plumbers Conference
+was that approach presented in "TCG PC Client Specific Implementation
+Specification for Conventional BIOS" is preferred. What in short means that
+BIOS should expose legacy interrupt INT 1Ah to legacy operating system for
+measurement purposes.
+
+There are couple problems with that:
+* this approach is complaint with TPM1.2, no support for TPM2.0
+* GRUB2 is neither consumer or producer of INT 1Ah
+* at this point GRUB2 supports TPM only through UEFI API
+* GRUB2 should preserve interrupt handlers
+
+Because whole problem seems to not be so important (it affects limited amount
+of hardware) we are thinking about simplest way of having correct results. It
+seems that solution that has most sense is leveraging [Rhode and Schwarz TrustedGRUB2 implementation](https://github.com/Rohde-Schwarz/TrustedGRUB2).
+
+Key assumption of Rhode and Schwarz project is that BIOS measures MBR and
+interrupts for measuring bootloader kernel (`diskboot.img`) is called from MBR.
+In that way measurement chain continues. Of course something has to install
+those interrupts, in this case it is SeaBIOS, which implemented previous
+mentioned TCG specification.
+
+Addressing lack of support for TPM2.0 through INT 1Ah seems to not be big
```

Comment:
Addressing a lack of support for TPM2.0 through INT 1Ah seems not to be a big issue since we can hide hardware that we deal with in seaBIOS implementation.